bach/bach/operations/cut.py

```diff
@@ -222,5 +224,118 @@ def _calculate_bucket_ranges(self, bucket_properties_df: 'DataFrame') -> 'Series
             ),
         )
         range_df.materialize(node_name='bin_ranges', inplace=True)
-
         return range_df.all_series[self.RANGE_SERIES_NAME]
+
+
+class QCutOperation:
+    """
+    In order to instantiate this class you should provide the following params:
+    series: A numerical series
+    q: The number of quantiles or list of quantiles to be calculated
+
+    returns a new Series containing the quantile ranges per each value on the series.
+
+    Example:
+        QCutOperation(s1, q=4)()   # will calculated quantiles `0, 0.25, 0.5, 0.75, 1`
+        or
+        QCutOperation(s1, q=[0.25, 0.5, 0.75])()
+    """
+
+    series: SeriesAbstractNumeric
+    quantiles: List[float]
+
+    RANGE_SERIES_NAME = 'q_range'
+
+    def __init__(self, series: SeriesAbstractNumeric, q: Union[int, List[float]]) -> None:
+        self.series = series
+        self.quantiles = q if isinstance(q, list) else numpy.linspace(0, 1, q + 1).tolist()
+
+    def __call__(self, *args, **kwargs) -> SeriesFloat64:
+        """
+        Gets the quantile range per bucket and assigns the correct range to each value. If the value
+        is not contained in any range, then it will be null.
+        on it, a
+        """
+        df = self.series.to_frame()
+        df.reset_index(drop=True, inplace=True)
+
+        if len(self.quantiles) == 1:
+            # need at least 2 quantiles for a range
+            df[self.RANGE_SERIES_NAME] = None
+        else:
+            quantile_ranges_df = self._get_quantile_ranges()
```

Comment:
```suggestion
            quantile_ranges = self._get_quantile_ranges()
```
the `_df` suffix is a bit misleading as this is a series, not a dataframe

Reply:
whoopps, thanks!